source/simulator/stokes_matrix_free.cc

```diff
@@ -1697,6 +1765,114 @@ namespace aspect
             level_cell_data[level].enable_newton_derivatives = false;
         }
     }
+
+    {
+      // Create active mesh tables to store the product of the pressure perturbation and
+      // the normalized gravity use in the free surface stabilization.
+
+      active_cell_data.apply_stabilization_free_surface_faces = false;
+
+      if (!sim.mesh_deformation->get_free_surface_boundary_indicators().empty())
+        {
+          const double free_surface_theta =
+            sim.mesh_deformation->template get_matching_mesh_deformation_object<MeshDeformation::FreeSurface<dim>>()
+          .get_free_surface_theta();
+
+          if (sim.parameters.include_melt_transport)
+            {
+              // GMG doesn't support melt transport
+              // It won't get here because of the assert in the constructor
+              AssertThrow(!sim.parameters.include_melt_transport, ExcNotImplemented());
```

Comment:
```suggestion
              // GMG doesn't support melt transport
              AssertThrow(!sim.parameters.include_melt_transport, ExcNotImplemented());
```
and just remove the whole ``if ... else``.